### ISSUE TITLE
Add support for HLS/DASH content steering

### ISSUE THREAD

**Issue by dchen-harmonicinc** [androidx/media#1689]
Content steering is a new feature of both HLS and DASH standards, simplifying the design of multi-CDN systems, redundancy/failover functions, load balancing, and various other traffic optimization applications. A number of players (DASH.js, HLS.js, Apple's AVplayer, etc.) already support this feature.

The relevant specifications are:
[1] HLS Content Steering Specification (v1.2b1) https://developer.apple.com/streaming/HLSContentSteeringSpecification.pdf
[2] DASH-IF CTS Version 0.9.0 – https://dashif.org/docs/DASH-IF-CTS-00XX-Content-Steering-Community-Review.pdf

The ask is: please add support for Content Steering functionality in Exoplayer.

Finer-level detail: if feasible, please include the _HLS/DASH_pathway and _HLS/DASH_throughput parameters in calls to the steering servers. These parameters help servers to react faster in cases of failures or other problems with CDNs.

**Comment by marcbaechinger** [comment]
Thanks for filing. 

I assign to @tonihei like issue https://github.com/google/ExoPlayer/issues/11203.. Usefulness of content steering for DASH/HLS is already acknowledged in that issue.

**Comment by tonihei** [comment]
Yes, this is a duplicate of https://github.com/google/ExoPlayer/issues/11203, but I'll use the opportunity to move the issue over to the Media3 tracker and close the old one. We are still planning to implement HLS and DASH content steering.

**Comment by yreznik-brightcove** [comment]
FYI, a recent overview paper describing various features and benefits of this standard: https://www.ibc.org/technical-papers/ibc2024-tech-papers-content-steering-a-standard-for-multi-cdn-streaming/12072.article It also includes references to various existing implementations, streams, and tools that may be useful for testing.

**Comment by joalag01** [comment]
FYI, we (a swedish public broadcaster) have tried content steering (using HLS) as a pilot with good results. It works well for both our web player (based on HLS.js) and iOS player (based on AVPlayer). 

If there is anything we can do to assist to implement this ticket, testing etc., please let us know.

**Comment by borisasadanin** [comment]
@tonihei or @marcbaechinger, I would be happy to to talk about how we can help to resolve this ticket. Let me know.
/Streaming Tech Specialists from Eyevinn

**Comment by gwendalsimon** [comment]
Is there any news about the integration of Content Steering?

**Comment by joalag01** [comment]
I see some encouraging commits, any plans for a release?

**Comment by chinbhat** [comment]
Hello, Any details you could provide on the DASH content steering support for Media3 ? All of the other major players support this functionality, it would be fantastic to get the Media3 to support content steering as well.

**Comment by chinbhat** [comment]
Thank you. Looks like support is being added for HLS. Any roadmap for DASH ?

**Comment by tianyif** [comment]
The HLS Content Steering is now available in our 1.11.0-alpha01 (https://github.com/androidx/media/tree/1.11.0-alpha01), we are working on the DASH Content Steering as well, which will be in our 1.12.0 release cycle.
